### PR TITLE
Feature/storybook controls

### PR DESCRIPTION
**Pull request by MKaHead**
Migration of storybook knobs for controls
- Using the latest API and args
- Remove redundant code
- Optimize stories
- Changes of story names
- Adjustment of test files
- Adapt and improve create new cp script

Disadvantages:
- I had to remove some tests because they are not yet compatible with storybook controls. What doesn't work yet escapeHTML: false this will probably come with the next version.
- Some tests were not compatible or flaky  I have deleted them and included them in this issue #2284.

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch-webhub-cloud/pattern-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -58,4 +56,11 @@ addParameters({
   viewport: {
     viewports: customViewports,
   },
-});
+  backgrounds: {
+    values: [
+      { name: 'red', value: '#f00' },
```
Comment (markus-walther) [inline .storybook/preview.js:61]: Why these 3 colours? And what does it do?

File: scripts/create-a-m-o/functions.js [hunk 2]
```diff
@@ -193,30 +193,70 @@ const createFiles = (store, a, m, o, done) => () => {
...
+    'utf8'
+  );
+
   fs.writeFileSync(
     `${BASE_FOLDER}/story.js`,
     outdent`
-    import { text, withKnobs } from '@storybook/addon-knobs';
-    import { html, render } from 'lit';
+    import { html } from 'lit';
+    import { args, argTypes } from './story.args';
```
Comment (markus-walther) [inline scripts/create-a-m-o/functions.js:243]: Nice that you updated the component generator!

File: scripts/create-a-m-o/functions.js [hunk 3]
```diff
@@ -193,30 +193,70 @@ const createFiles = (store, a, m, o, done) => () => {
...
-      const wrapper = document.createElement('div');
-      const template = html\`
-        <axa-${fileName}>\${textknob}</axa-${fileName}>
-      \`;
-      
-      render(template, wrapper);
-      return wrapper;
-    }
-    `,
+    export const ${compTitle} = ({ slot }) => html\`<axa-${fileName}>\${slot}</axa-${fileName}>\`;`,
```
Comment (markus-walther) [inline scripts/create-a-m-o/functions.js:259]: Nice, much easier!

File: src/components/10-atoms/checkbox/demo.js [hunk 4]
```diff
@@ -1,63 +1,48 @@
...
+`;
 
-    const wrapper = document.createElement('div');
-    render(template, wrapper);
-    return wrapper;
-  });
+export const WithoutLabel = () => html`
+  <axa-checkbox
+    name="checkbox"
+    error="This checkbox do not have a label, but its also clickable"
```
Comment (markus-walther) [inline src/components/10-atoms/checkbox/demo.js]: Grammar: .... do**es** not have a label, but **it's still** clickable

File: src/components/10-atoms/input-text/react/story.jsx [hunk 5]
```diff
@@ -127,5 +46,19 @@ storyInputText.add('Story - Simulate autocomplete', () => {
     }
   }, 1);
 
-  return container;
-});
+  // there are two input fields because with only one is not possible to
+  // activate the Safari's autocomplete feature.
```
Comment (markus-walther) [inline src/components/10-atoms/input-text/react/story.jsx]: grammar: activate Safari's autocomplete feature (**no _the_**)

File: src/components/10-atoms/input-text/react/story.jsx [hunk 6]
```diff
@@ -1,119 +1,38 @@
...
+      value={value}
+      onChange={evt => setValue(evt.target.value)}
+    />
   );
+};
 
+export const InputText = _args =>
+  createReactContainer(<InputTextControlled {..._args} />);
+
+/* TODO rewrite this
```
Comment (markus-walther) [inline src/components/10-atoms/input-text/react/story.jsx:34]: What does that mean here?

File: src/components/10-atoms/textarea/demo.js [hunk 7]
```diff
@@ -34,48 +32,23 @@ storiesOf('Examples/Textarea/Pure HTML', module)
...
-    setTimeout(() => {
-      document
-        .querySelector('axa-textarea')
-        .setAttribute('counter', 'Still ##counter## left');
-      document.querySelector('axa-textarea').setAttribute('maxLength', 100);
-      document
-        .querySelector('axa-textarea')
-        .setAttribute('countermax', 'Max maxLength reached');
-    }, 2000);
+/* TODO rewrite this
```
Comment (markus-walther) [inline src/components/10-atoms/textarea/demo.js:40]: This looks ugly, commented-out stuff should not be retained - version control takes care of earlier revisions, not comments.

File: src/components/10-atoms/textarea/react/DemoControlledUncontrolledTextarea.jsx [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import React, { useState } from 'react';
 import AXATextarea from './AXATextarea';
 
-const DemoContUncontTextarea = () => {
+const DemoControlledUncontrolledTextarea = () => {
```
Comment (markus-walther) [inline src/components/10-atoms/textarea/react/DemoControlledUncontrolledTextarea.jsx]: Very confusing name. `DemoControllabilityOfTextarea` perhaps?

File: src/components/10-atoms/textarea/react/demo.jsx [hunk 9]
```diff
@@ -1,21 +1,18 @@
-import { storiesOf } from '@storybook/web-components';
 import React from 'react';
-import { createRoot } from 'react-dom/client';
 import changelog from '../CHANGELOG.md';
 import readme from '../README.md';
-import DemoContUncontTextarea from './DemoContUncontTextarea';
+import DemoControlledUncontrolledTextarea from './DemoControlledUncontrolledTextarea';
```
Comment (markus-walther) [inline src/components/10-atoms/textarea/react/demo.jsx]: see above for naming

File: src/components/10-atoms/textarea/react/demo.jsx [hunk 10]
```diff
@@ -1,21 +1,18 @@
...
-    const container = document.createElement('div');
-    const root = createRoot(container);
-    root.render(<DemoContUncontTextarea />);
+    controls: { disable: true },
+  },
+};
 
-    return container;
-  });
+export const TextareaControlledUncontrolled = () =>
```
Comment (markus-walther) [inline src/components/10-atoms/textarea/react/demo.jsx]: see above for naming

File: src/utils/create-react-container.jsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createRoot } from 'react-dom/client';
+
+export function createReactContainer(
```
Comment (markus-walther) [inline src/utils/create-react-container.jsx]: why function? My default would have been to use arrow function wherever possible...

File: src/utils/render-lit-container.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import { render } from 'lit';
+
+export function renderLitContainer(
```
Comment (markus-walther) [inline src/utils/render-lit-container.js]: ditto

**Review by markus-walther — CHANGES_REQUESTED** [review]
Dear Marco,

a ton of good work - a challenge to review, of course, due to the bulk changes you had to make.

Few things that caught my eye, try to do the best with them....

Keep up the good work, Markus

File: .storybook/preview.js (hunk 1 above)
Comment (MKaHead) [reply .storybook/preview.js:61]: For example, there are components that have a variant where their background becomes white. Then the problem is that you can't see the component at all. Some stories had functionality to switch background. With the change I remove redundancies and provide further uniformity.

Comment (markus-walther) [reply .storybook/preview.js:61]: I see, that's OK.

**Pull request merged**